### PR TITLE
add aws_string_compare function

### PR DESCRIPTION
**Pull request by david-koenig**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/string.h [hunk 1]
```diff
@@ -57,6 +57,19 @@ AWS_COMMON_API const struct aws_string *aws_string_from_array_new(struct aws_all
  */
 AWS_COMMON_API void aws_string_destroy(void *str);
 
+/**
+ * Compares lexicographical ordering of two strings. If both strings are identical in the
```
Comment (bdonlan) [inline include/aws/common/string.h]: Is this a binary comparison, or a unicode-aware comparison?

File: include/aws/common/string.h [hunk 2]
```diff
@@ -57,6 +57,19 @@ AWS_COMMON_API const struct aws_string *aws_string_from_array_new(struct aws_all
...
+ * bytes of the shorter string, then the longer string is lexicographically after the shorter.
+ *
+ * Returns a positive if string a > string b. (i.e., string a is lexicographically after string b.)
```
Comment (bdonlan) [inline include/aws/common/string.h]: a positive number

File: include/aws/common/string.h [hunk 3]
```diff
@@ -57,6 +57,19 @@ AWS_COMMON_API const struct aws_string *aws_string_from_array_new(struct aws_all
...
+ * Returns zero if string a = string b.
+ * Returns negative if string a < string b.
```
Comment (bdonlan) [inline include/aws/common/string.h]: a negative number

File: tests/string_test.c [hunk 4]
```diff
@@ -81,3 +81,23 @@ static int binary_string_test_fn(struct aws_allocator *alloc, void *ctx) {
     aws_string_destroy((void *)binary_string);
     return 0;
 }
+
+AWS_TEST_CASE(string_compare_test, string_compare_test_fn);
+static int string_compare_test_fn(struct aws_allocator *alloc, void *ctx) {
+    AWS_STATIC_STRING_FROM_LITERAL(str_1, "foo");
```
Comment (bdonlan) [inline tests/string_test.c]: name these strings after their contents to make it easier to scan the test cases

File: tests/string_test.c [hunk 5]
```diff
@@ -81,3 +81,23 @@ static int binary_string_test_fn(struct aws_allocator *alloc, void *ctx) {
...
+    AWS_STATIC_STRING_FROM_LITERAL(str_5, "foobaz");
+    AWS_STATIC_STRING_FROM_LITERAL(str_6, "bar food");
+
+    ASSERT_TRUE(aws_string_compare(str_1, str_2) > 0);
+    ASSERT_TRUE(aws_string_compare(str_2, str_1) < 0);
+    ASSERT_TRUE(aws_string_compare(str_1, str_3) < 0);
+    ASSERT_TRUE(aws_string_compare(str_1, str_4) == 0);
+    ASSERT_TRUE(aws_string_compare(str_3, str_5) < 0);
+    ASSERT_TRUE(aws_string_compare(str_1, str_6) > 0);
+    ASSERT_TRUE(aws_string_compare(str_6, str_2) > 0);
```
Comment (bdonlan) [inline tests/string_test.c]: Have some tests for strings with NULs. Also test the empty-string case.
